diff --git a/ckan/cli/server.py b/ckan/cli/server.py
index 84f209c2c..4f461d7ac 100644
--- a/ckan/cli/server.py
+++ b/ckan/cli/server.py
@@ -4,6 +4,7 @@ import logging

 import click
 from werkzeug.serving import run_simple
+from werkzeug.middleware.dispatcher import DispatcherMiddleware

 import ckan.plugins.toolkit as tk
 from ckan.common import config
@@ -39,9 +40,13 @@ DEFAULT_PORT = 5000
     u"-K", u"--ssl-key", default=None,
     help=u"Key file to use to enable SSL. Passing 'adhoc' will "
     " automatically generate a new one (on each server reload).")
+@click.option(
+    u"-P", u"--prefix", default="",
+    help=u"Run ckan in prefix path."
+)
 @click.pass_context
 def run(ctx, host, port, disable_reloader, threaded, extra_files, processes,
-        ssl_cert, ssl_key):
+        ssl_cert, ssl_key, prefix):
     u"""Runs the Werkzeug development server"""

     # Reloading
@@ -74,6 +79,15 @@ def run(ctx, host, port, disable_reloader, threaded, extra_files, processes,
     else:
         ssl_context = None

+    if prefix:
+        if not prefix.startswith(u'/'):
+            tk.error_shout(u"Prefix must start with /, example /data.")
+            raise click.Abort()
+        ctx.obj.app = DispatcherMiddleware(ctx.obj.app, {
+            prefix: ctx.obj.app
+        })
+
+
     host = host or config.get(u'ckan.devserver.host', DEFAULT_HOST)
     port = port or config.get(u'ckan.devserver.port', DEFAULT_PORT)
     try:
@@ -82,8 +96,9 @@ def run(ctx, host, port, disable_reloader, threaded, extra_files, processes,
         tk.error_shout(u"Server port must be an integer, not {}".format(port))
         raise click.Abort()
 
-    log.info(u"Running CKAN on {scheme}://{host}:{port}".format(
-        scheme=u"https" if ssl_context else u"http", host=host, port=port))
+    log.info(u"Running CKAN on {scheme}://{host}:{port}{prefix}".format(
+        scheme=u"https" if ssl_context else u"http", host=host, port=port,
+        prefix=prefix))
 
     run_simple(
         host,
