diff --git a/ckan/lib/search/query.py b/ckan/lib/search/query.py
index c2646f868..d928722b3 100644
--- a/ckan/lib/search/query.py
+++ b/ckan/lib/search/query.py
@@ -268,7 +268,7 @@ class PackageSearchQuery(SearchQuery):
             'rows': 1,
             'q': 'name:"%s" OR id:"%s"' % (reference,reference),
             'wt': 'json',
-            'fq': 'site_id:"%s"' % config.get('ckan.site_id')}
+            'fq': 'site_id:"%s" ' % config.get('ckan.site_id') + '+entity_type:package'}
 
         try:
             if query['q'].startswith('{!'):
