diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index 4858d7063..d9b0e525b 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -2660,7 +2660,7 @@ def load_plugin_helpers() -> None:
     helper_functions.update(_builtin_functions)
     chained_helpers = defaultdict(list)

-    for plugin in p.PluginImplementations(p.ITemplateHelpers):
+    for plugin in reversed(list(p.PluginImplementations(p.ITemplateHelpers))):
         for name, func in plugin.get_helpers().items():
             if _is_chained_helper(func):
                 chained_helpers[name].append(func)
