diff --git a/ckan/plugins/toolkit.py b/ckan/plugins/toolkit.py
index 83a6d2056e..ba1c6a2798 100644
--- a/ckan/plugins/toolkit.py
+++ b/ckan/plugins/toolkit.py
@@ -194,12 +194,10 @@ def add_public_directory(config_: CKANConfig, relative_path: str):
     public urls.
 
     """
-    from ckan.lib.helpers import _local_url
     from ckan.lib.webassets_tools import add_public_path
 
     path = _add_served_directory(config_, relative_path, "plugin_public_paths")
-    url = _local_url("/", locale="default")
-    add_public_path(path, url)
+    add_public_path(path, '/')
 
 
 def _add_served_directory(
