diff --git a/ckan/lib/base.py b/ckan/lib/base.py
index dcbfc2db9..dfa94fac6 100644
--- a/ckan/lib/base.py
+++ b/ckan/lib/base.py
@@ -28,7 +28,7 @@ from ckan.views import (identify_user,
                         check_session_cookie,
                         )
 from ckan.common import (c, request, config,
-                         session, is_flask_request, asbool)
+                         session, is_flask_request, asbool, g)
 
 
 if six.PY2:
@@ -239,7 +239,7 @@ def _allow_caching(cache_force=None):
     if cache_force is not None:
         allow_cache = cache_force
     # Do not allow caching of pages for logged in users/flash messages etc.
-    elif _is_valid_session_cookie_data():
+    elif ('user' in g and g.user) or _is_valid_session_cookie_data():
         allow_cache = False
     # Tests etc.
     elif 'REMOTE_USER' in request.environ:
