diff --git a/drush_language.drush.inc b/drush_language.drush.inc
index 76d8135..54cce15 100644
--- a/drush_language.drush.inc
+++ b/drush_language.drush.inc
@@ -140,7 +140,7 @@ function drush_drush_language_language_add() {
 /**
  * Enable a language
  */
-function drush_language_enable() {
+function drush_drush_language_language_enable() {
   $args = func_get_args();
   if (count($args) == 0) {
     drush_set_error(dt('Please provide one or more language codes as arguments.'));
@@ -176,7 +176,7 @@ function drush_language_enable() {
 /**
  * Disables a language
  */
-function drush_drush_language_disable() {
+function drush_drush_language_language_disable() {
   $args = func_get_args();
   if (count($args) == 0) {
     drush_set_error(dt('Please provide one or more language codes as arguments.'));
@@ -210,7 +210,7 @@ function drush_drush_language_disable() {
 /**
  * Assigns the default language
  */
-function drush_drush_language_default() {
+function drush_drush_language_language_default() {
   $args = func_get_args();
   if (count($args) == 0) {
     drush_set_error(dt('Please provide one or more language codes as arguments.'));
@@ -222,9 +222,7 @@ function drush_drush_language_default() {
     $language_manager = \Drupal::languageManager();
     $languages = $language_manager->getLanguages();
     if (array_key_exists($langcode, $languages)) {
-      $default_language = ConfigurableLanguage::load($langcode);
-      $default_language->set('default', TRUE);
-      $default_language->save();
+      \Drupal::configFactory()->getEditable('system.site')->set('default_langcode', $langcode)->save();
       drush_log(dt("!language assigned as default", array('!language' => $langcode)), 'ok');
     }
     else {