diff --git a/ckan/config/middleware/flask_app.py b/ckan/config/middleware/flask_app.py
index 368bbaf22..4fb412b21 100644
--- a/ckan/config/middleware/flask_app.py
+++ b/ckan/config/middleware/flask_app.py
@@ -6,6 +6,8 @@ import inspect
 import itertools
 import pkgutil

+import autodynatrace
+
 from flask import Flask, Blueprint
 from flask.ctx import _AppCtxGlobals
 from flask.sessions import SessionInterface
