diff --git a/ckan/model/activity.py b/ckan/model/activity.py
index b56d91e..6f3377f 100644
--- a/ckan/model/activity.py
+++ b/ckan/model/activity.py
@@ -198,14 +198,12 @@ def _group_activity_query(group_id):
         model.Member,
         and_(
             model.Activity.object_id == model.Member.table_id,
-            model.Member.state == 'active'
         )
     ).outerjoin(
         model.Package,
         and_(
             model.Package.id == model.Member.table_id,
             model.Package.private == False,
-            model.Package.state == 'active'
         )
     ).filter(
         # We only care about activity either on the the group itself or on
@@ -214,9 +212,19 @@ def _group_activity_query(group_id):
         # to a group but was then removed will not show up. This may not be
         # desired but is consistent with legacy behaviour.
         or_(
-            model.Member.group_id == group_id,
-            model.Activity.object_id == group_id
-        ),
+            # active dataset in the group
+            and_(model.Member.group_id == group_id,
+                 model.Member.state == 'active',
+                 model.Package.state == 'active'),
+            # deleted dataset in the group
+            and_(model.Member.group_id == group_id,
+                 model.Member.state == 'deleted',
+                 model.Package.state == 'deleted'),
+                 # (we want to avoid showing changes to an active dataset that
+                 # was once in this group)
+            # activity the the group itself
+            model.Activity.object_id == group_id,
+        )
     )
 
     return q
