diff --git a/ckan/controllers/group.py b/ckan/controllers/group.py
index 6aa3d1f77..3f58e8683 100644
--- a/ckan/controllers/group.py
+++ b/ckan/controllers/group.py
@@ -220,6 +220,10 @@ class GroupController(base.BaseController):
             # Do not query for the group datasets when dictizing, as they will
             # be ignored and get requested on the controller anyway
             data_dict['include_datasets'] = False
+
+            # Do not query group members as they aren't used in the view
+            data_dict['include_users'] = False
+
             c.group_dict = self._action('group_show')(context, data_dict)
             c.group = context['group']
         except (NotFound, NotAuthorized):
