diff --git a/ckan/migration/versions/087_create_index_group_extra_group_id.py b/ckan/migration/versions/087_create_index_group_extra_group_id.py
new file mode 100644
index 000000000..7311bb0a6
--- /dev/null
+++ b/ckan/migration/versions/087_create_index_group_extra_group_id.py
@@ -0,0 +1,10 @@
+from sqlalchemy import *
+from migrate import *
+
+
+def upgrade(migrate_engine):
+  migrate_engine.execute('CREATE INDEX idx_group_extra_group_id ON group_extra (group_id);')
+
+
+def downgrade(migrate_engine):
+  migrate_engine.execute('DROP INDEX idx_group_extra_group_id;')
diff --git a/ckan/migration/versions/088_create_index_resource_package_id.py b/ckan/migration/versions/088_create_index_resource_package_id.py
new file mode 100644
index 000000000..b660f61af
--- /dev/null
+++ b/ckan/migration/versions/088_create_index_resource_package_id.py
@@ -0,0 +1,10 @@
+from sqlalchemy import *
+from migrate import *
+
+
+def upgrade(migrate_engine):
+  migrate_engine.execute('CREATE INDEX idx_resource_package_id ON resource (package_id);')
+
+
+def downgrade(migrate_engine):
+  migrate_engine.execute('DROP INDEX idx_resource_package_id;')
