diff --git a/ckan/views/__init__.py b/ckan/views/__init__.py
index b39fef0fe..a0057d051 100644
--- a/ckan/views/__init__.py
+++ b/ckan/views/__init__.py
@@ -54,7 +54,7 @@ def set_cors_headers_for_response(response):
 
         cors_origin_allowed = None
         if asbool(config.get(u'ckan.cors.origin_allow_all')):
-            cors_origin_allowed = u'*'
+            cors_origin_allowed = b'*'
         elif config.get(u'ckan.cors.origin_whitelist') and \
                 request.headers.get(u'Origin') \
                 in config[u'ckan.cors.origin_whitelist'].split(u' '):
@@ -62,12 +62,12 @@ def set_cors_headers_for_response(response):
             cors_origin_allowed = request.headers.get(u'Origin')
 
         if cors_origin_allowed is not None:
-            response.headers[u'Access-Control-Allow-Origin'] = \
+            response.headers[b'Access-Control-Allow-Origin'] = \
                 cors_origin_allowed
-            response.headers[u'Access-Control-Allow-Methods'] = \
-                u'POST, PUT, GET, DELETE, OPTIONS'
-            response.headers[u'Access-Control-Allow-Headers'] = \
-                u'X-CKAN-API-KEY, Authorization, Content-Type'
+            response.headers[b'Access-Control-Allow-Methods'] = \
+                b'POST, PUT, GET, DELETE, OPTIONS'
+            response.headers[b'Access-Control-Allow-Headers'] = \
+                b'X-CKAN-API-KEY, Authorization, Content-Type'
 
     return response
 
