diff --git a/ckan/views/group.py b/ckan/views/group.py
index dd38fce8c..9ea566e1e 100644
--- a/ckan/views/group.py
+++ b/ckan/views/group.py
@@ -923,7 +923,7 @@ class EditGroupView(MethodView):
 
         try:
             group = _action(u'group_show')(context, data_dict)
-            check_access(u'group_update', context)
+            _check_access(u'group_update', context, {u'id': id})
         except NotAuthorized:
             base.abort(403, _(u'Unauthorized to create a group'))
         except NotFound:
