From e7936fe8707a847387661248b5b0d9d3ddc526ae Mon Sep 17 00:00:00 2001
From: Ville Seppanen <ville.seppanen@gofore.com>
Date: Tue, 25 Mar 2014 15:26:58 +0200
Subject: [PATCH] YH-272: Added support for Finnish and Swedish languages

---
 ckanext/multilingual/plugin.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ckanext/multilingual/plugin.py b/ckanext/multilingual/plugin.py
index 9a519b4..7e0183f 100644
--- a/ckanext/multilingual/plugin.py
+++ b/ckanext/multilingual/plugin.py
@@ -7,7 +7,7 @@ from ckan.plugins import IGroupController, IOrganizationController, ITagControll
 from ckan.common import request, config, c
 from ckan.logic import get_action

-LANGS = ['en', 'fr', 'de', 'es', 'it', 'nl', 'ro', 'pt', 'pl']
+LANGS = ['en', 'fi', 'fr', 'de', 'es', 'it', 'nl', 'ro', 'pt', 'pl', 'sv']

 def translate_data_dict(data_dict):
     '''Return the given dict (e.g. a dataset dict) with as many of its fields
