diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index a701ef5ae..dc87b5fd5 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -152,6 +152,12 @@ def redirect_to(*args, **kw):
 
         toolkit.redirect_to('dataset_read', id='changed')
 
+    If given a single string as argument, this redirects without url parsing
+
+        toolkit.redirect_to('http://example.com')
+        toolkit.redirect_to('/dataset')
+        toolkit.redirect_to('/some/other/path')
+
     '''
     if are_there_flash_messages():
         kw['__no_cache__'] = True
@@ -159,7 +165,24 @@ def redirect_to(*args, **kw):
     # Routes router doesn't like unicode args
     uargs = map(lambda arg: str(arg) if isinstance(arg, unicode) else arg,
                 args)
-    _url = url_for(*uargs, **kw)
+
+    # Repoze.who redirects with single strings
+    # which need to be parsed by url_for
+    exempt_urls = ['/user/logout', '/user/logged_out_redirect',
+                   '/user/logged_in']
+    matching = [s for s in uargs if any(xs in s for xs in exempt_urls)]
+
+    _url = ''
+    skip_url_parsing = False
+    if uargs and len(uargs) is 1 and isinstance(uargs[0], basestring) \
+            and (uargs[0].startswith('/') or is_url(uargs[0])) \
+            and len(matching) is 0:
+        skip_url_parsing = True
+        _url = uargs[0]
+
+    if skip_url_parsing is False:
+        _url = url_for(*uargs, **kw)
+
     if _url.startswith('/'):
         _url = str(config['ckan.site_url'].rstrip('/') + _url)
 
