diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index 0d9a2ed8c..38d753c49 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -1503,17 +1503,7 @@ _VALID_GRAVATAR_DEFAULTS = ['404', 'mm', 'identicon', 'monsterid',
 
 @core_helper
 def gravatar(email_hash, size=100, default=None):
-    if default is None:
-        default = config.get('ckan.gravatar_default', 'identicon')
-
-    if default not in _VALID_GRAVATAR_DEFAULTS:
-        # treat the default as a url
-        default = quote(default, safe='')
-
-    return literal('''<img src="//gravatar.com/avatar/%s?s=%d&amp;d=%s"
-        class="user-image" width="%s" height="%s" alt="Gravatar" />'''
-                   % (email_hash, size, default, size, size)
-                   )
+    return ''
 
 
 _PLAUSIBLE_HOST_IDNA = re.compile(r'^[-\w.:\[\]]*$')
